micrometer-spring-legacy/src/main/java/io/micrometer/spring/web/client/DefaultRestTemplateExchangeTagsProvider.java

```diff
@@ -32,9 +32,9 @@
     implements RestTemplateExchangeTagsProvider {
 
     @Override
-    public Iterable<Tag> getTags(String urlTemplate, HttpRequest request,
+    public Iterable<Tag> getTags(@Nullable String urlTemplate, HttpRequest request,
                                  @Nullable ClientHttpResponse response) {
-        Tag uriTag = StringUtils.hasText(urlTemplate)
+        Tag uriTag = urlTemplate != null && StringUtils.hasText(urlTemplate)
```

Comment:
Do we still need to check `StringUtils.hasText(urlTemplate)`?

Reply:
`StringUtils.hasText` checks for null and returns `false` if the input is `null`. The tests still pass without the additional `urlTemplate != null` check. Not sure we have captured the case that yields an NPE?

Reply:
I figured just `urlTemplate != null` without the has text check would be preferred but didn't know if there was a case where we would still want to use the request URI if `urlTemplate` is blank.

Reply:
It does at least remove an IDE warning since `hasText` doesn't have a `@Nullable` marker until Spring 5.